index.js

```diff
@@ -2613,6 +2613,42 @@
   }
   S.ifElse = def('ifElse', {}, [Pred(a), Fn(a, b), Fn(a, b), a, b], ifElse);
 
+  //# when :: (a -> Boolean) -> (a -> b) -> a -> b
```

Comment:
Signature is not quite correct.

We have:
```haskell
when :: (a -> Boolean) -> (a -> b) -> a -> b
```

If second function is for example from Int to String then you can't give back String in if predicate fails:
```js
// :: (Int -> Boolean) -> (Int -> String) -> Int -> String
when(a => a < 0,ai => `got ${a}`, -1) // will get error as `Int` is not `String`
```